### DISCUSSION TITLE
Edge Functions: Where to store third party API refresh tokens that are not related to a user?

### DISCUSSION THREAD

**Discussion by felixhagspiel** [supabase/supabase#14865] [Questions] [answered]
I have an edge function that serves as a user endpoint, which contacts a third party API on every request. The third party API requires to get an refresh token by using client ID and secret. client ID and secret are stored in the environment variables. As the third party API has rate limits, I want to avoid getting an refresh token for each user request, when an older refresh token might still be valid.

Is there any recommended (safe) way to store the refresh token to share it between requests? Last possible solution would be to create an encrypted key / value table in supabase and to store it there, which I would like to avoid.

**Comment by j4w8n** [comment, marked as answer]
I haven't tested this out, but I'm thinking you could use the Vault and create any setter/getter functions you need. Then from your edge function use rpc to call those functions with a supabase client, auth'd with the `service_role` key.

**Comment by bombillazo** [comment]
Supabase doesnt have anything like a Redis or memcache instance were these would be ideally stored, but perhaps the new Vault module is worth a look, although it is analogous to what you are trying to avoid.
